File: src/main/java/com/pedigree/model/ProjectLayout.java
package com.pedigree.model;

import java.util.HashMap;
import java.util.Map;

public class ProjectLayout {
    public static class NodePos {
        public double x;
        public double y;
    }

    private double zoom = 1.0;
    private double viewOriginX = 0.0;
    private double viewOriginY = 0.0;
    private final Map<String, NodePos> nodePositions = new HashMap<>();
    // If true, nodePositions are centers; UI should convert to top-left once and then reset this flag
    private boolean positionsAreCenters = false;

    public double getZoom() { return zoom; }
    public void setZoom(double zoom) { this.zoom = zoom; com.pedigree.util.DirtyFlag.setModified(); }
    public double getViewOriginX() { return viewOriginX; }
    public void setViewOriginX(double viewOriginX) { this.viewOriginX = viewOriginX; com.pedigree.util.DirtyFlag.setModified(); }
    public double getViewOriginY() { return viewOriginY; }
    public void setViewOriginY(double viewOriginY) { this.viewOriginY = viewOriginY; com.pedigree.util.DirtyFlag.setModified(); }
    public Map<String, NodePos> getNodePositions() { return nodePositions; }

    public boolean isPositionsAreCenters() { return positionsAreCenters; }
    public void setPositionsAreCenters(boolean positionsAreCenters) { this.positionsAreCenters = positionsAreCenters; com.pedigree.util.DirtyFlag.setModified(); }
}

File: src/main/kotlin/com/family/tree/model/ProjectLayout.kt
package com.family.tree.model

import com.family.tree.util.DirtyFlag
import com.fasterxml.jackson.annotation.JsonAlias
import com.fasterxml.jackson.annotation.JsonProperty

class ProjectLayout {
    class NodePos {
        var x: Double = 0.0
        var y: Double = 0.0
    }

    var zoom: Double = 1.0
        set(value) { field = value; DirtyFlag.setModified() }

    var viewOriginX: Double = 0.0
        set(value) { field = value; DirtyFlag.setModified() }

    var viewOriginY: Double = 0.0
        set(value) { field = value; DirtyFlag.setModified() }

    val nodePositions: MutableMap<String, NodePos> = HashMap()

    /** If true, nodePositions are centers; UI should convert to top-left once and then reset this flag */
    @JsonProperty("positionsAreCenters")
    @JsonAlias("isPositionsAreCenters")
    var positionsAreCenters: Boolean = false
        set(value) { field = value; DirtyFlag.setModified() }

    // Java interop compatibility for boolean bean getter
    fun isPositionsAreCenters(): Boolean = positionsAreCenters
}
